File: CSharpFundamentals/14 Strings_Dictionaries_LINQ/URLParser/URLParser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace URLParser
{
    class URLParser
    {
        static void Main(string[] args)
        {
            string input = Console.ReadLine();
            string protocol = "";
            string server = "";
            string resource = "";
            int index = input.IndexOf("://");
            if (index > 0)
            {
                for (int i = 0; i < index; i++)
                {
                    protocol += input[i];
                }
                input = input.Remove(0, index + 3);
            }          
            index = input.IndexOf("/");
            if (index > 0)
            {
                for (int i = 0; i < index; i++)
                {
                    server += input[i];
                }
                input = input.Remove(0, index + 1);
                for (int i = 0; i < input.Length; i++)
                {
                    resource += input[i];
                }
            }
            else
            {
                for (int i = 0; i < input.Length; i++)
                {
                    server += input[i];
                }
            }
            Console.WriteLine(@"[protocol] = ""{0}""", protocol);
            Console.WriteLine(@"[server] = ""{0}""", server);
            Console.WriteLine(@"[resource] = ""{0}""", resource);
        }
    }
}

File: SoftwareTechnologies/Java/JavaBasics/src/URLParser.java
import java.util.Scanner;

public class URLParser {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        String url = scan.nextLine();

        String protocol = "";
        String server = "";
        String resource = "";

        int index = url.indexOf("://");
        if (index >= 0) {
            for (int i = 0; i < index; i++) {
                protocol += url.charAt(i);
            }
            url = url.substring(index + 3, url.length());
        }
        index = url.indexOf("/");
        if (index >= 0) {
            for (int i = 0; i < index; i++) {
                server += url.charAt(i);
            }
            for (int i = index + 1; i < url.length(); i++) {
                resource += url.charAt(i);
            }
        }
        else {
            for (int i = 0; i < url.length(); i++) {
                server += url.charAt(i);
            }
        }
        System.out.printf("[protocol] = \"%s\"\n", protocol);
        System.out.printf("[server] = \"%s\"\n", server);
        System.out.printf("[resource] = \"%s\"\n", resource);
    }
}
